tests/cpp/test_remove_bcast_squeeze.cpp

```diff
@@ -0,0 +1,322 @@
+  EXPECT_FALSE(has_bcast);
+  EXPECT_TRUE(has_squeeze);
+}
+
+TEST_F(RemoveBcastSqueezeTest, SqueezeBcastSqueezeBcast) {
+  auto fusion = std::make_unique<Fusion>();
+  FusionGuard fg(fusion.get());
+  DataType input_dtype = DataType::Float;
+  const std::vector<bool> is_broadcast_dim{false, true};
+  auto tv0 = makeBroadcastTensor(is_broadcast_dim, input_dtype);
+  fusion->addInput(tv0);
+  auto tv1 = set(tv0);
+  auto tv2 = squeeze(tv1, is_broadcast_dim);
+  auto tv3 = broadcast(tv2, is_broadcast_dim);
+  auto tv4 = squeeze(tv3, is_broadcast_dim);
+  auto tv5 = broadcast(tv4, is_broadcast_dim);
+  auto tv6 = set(tv5);
+  fusion->addOutput(tv6);
+
+  // squeeze output is the fusion output
+  // preseg_passes should remove broadcast but not squeeze
+  preseg_passes::OptimizationPass<preseg_passes::PreSegmenter>::runPass(
+      fusion.get());
+  bool has_bcast = ir_utils::hasOpsOfType<BroadcastOp>(fusion.get());
+  bool has_squeeze = ir_utils::hasOpsOfType<SqueezeOp>(fusion.get());
+  EXPECT_FALSE(has_bcast);
+  EXPECT_FALSE(has_squeeze);
+}
+
+TEST_F(RemoveBcastSqueezeTest, SqueezeBcastBcastSqueeze) {
+  auto fusion = std::make_unique<Fusion>();
+  FusionGuard fg(fusion.get());
+  DataType input_dtype = DataType::Float;
+  const std::vector<bool> is_broadcast_dim_1{false, true};
+  const std::vector<bool> is_broadcast_dim_2{false, false, true};
+  auto tv0 = makeBroadcastTensor(is_broadcast_dim_1, input_dtype);
+  fusion->addInput(tv0);
+  auto tv1 = set(tv0);
+  auto tv2 = squeeze(tv1, is_broadcast_dim_1);
+  auto tv3 = broadcast(tv2, is_broadcast_dim_1);
+  auto tv4 = set(tv3);
+  auto tv5 = broadcast(tv4, is_broadcast_dim_2);
+  auto tv6 = squeeze(tv5, is_broadcast_dim_2);
+  auto tv7 = set(tv6);
+  fusion->addOutput(tv7);
+
+  // preseg_passes should remove both broadcast and squeeze
+  preseg_passes::OptimizationPass<preseg_passes::PreSegmenter>::runPass(
+      fusion.get());
+  bool has_bcast = ir_utils::hasOpsOfType<BroadcastOp>(fusion.get());
+  bool has_squeeze = ir_utils::hasOpsOfType<SqueezeOp>(fusion.get());
+  EXPECT_FALSE(has_bcast);
+  EXPECT_FALSE(has_squeeze);
+}
+
+TEST_F(RemoveBcastSqueezeTest, BcastSqueezeBcast) {
+  auto fusion = std::make_unique<Fusion>();
+  FusionGuard fg(fusion.get());
+  DataType input_dtype = DataType::Float;
+  auto tv0 = makeContigTensor(2, input_dtype);
+  fusion->addInput(tv0);
+  auto tv1 = set(tv0);
+  auto tv2 = broadcast(tv1, {false, false, true});
+  auto tv3 = squeeze(tv2, std::vector<bool>{false, false, true});
+  auto tv4 = broadcast(tv3, {false, false, true});
+  auto tv5 = set(tv4);
+  fusion->addOutput(tv5);
+
+  // preseg_passes should remove the first broadcast and squeeze
+  preseg_passes::OptimizationPass<preseg_passes::PreSegmenter>::runPass(
+      fusion.get());
+  bool has_bcast = ir_utils::hasOpsOfType<BroadcastOp>(fusion.get());
+  bool has_squeeze = ir_utils::hasOpsOfType<SqueezeOp>(fusion.get());
+  EXPECT_TRUE(has_bcast);
+  EXPECT_FALSE(has_squeeze);
```

Comment:
```suggestion
  EXPECT_TRUE(ir_utils::hasOpsOfType<BroadcastOp>(fusion.get()));
  EXPECT_FALSE(ir_utils::hasOpsOfType<SqueezeOp>(fusion.get()));
```

The call to hasOpsOfType is pretty clear about its intention, so a local variable `has_bcast` or `has_squeeze` doesn't seem worthwhile.

Reply:
revised.